packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -331,10 +338,28 @@ export interface ICompressionRuntimeOptions {
 
 /**
  * Options for container runtime.
+ *
+ * @privateRemarks If any new properties are added to this interface (or IContainerRuntimeOptionsInternal), then we will also need
+ * to make changes in compatUtils.ts.
+ * If the new property is not considered version-dependent, then it must be explicity omitted from IContainerRuntimeOptionsVersionDependent.
+ * If it is considered version-dependent, then a corresponding entry must be added to `versionDependentOptionConfigMap` with the appropriate compat
+ * configuration info.
+ *
  * @legacy
  * @alpha
  */
 export interface IContainerRuntimeOptions {
+	/**
+	 * compatibilityMode is used to determine the default configuration for version-dependent options (see {@link IContainerRuntimeOptionsVersionDependent}
```

Comment:
The explanation you have on that type is reasonable, but this link won't resolve from the perspective of the alpha+legacy api: `IContainerRuntimeOptionsVersionDependent` is internal. I think the right fix is to just move content around so that we explain what version-dependent options are here (or somewhere else in the legacy+alpha API, I think making it obvious to consumers which of the options they change have compatibility implications is a net good thing) and reference it there.